README.md

```diff
@@ -26,6 +29,200 @@ to run `zosma`.
+supply an identity (getters omitted for brevity):
+
+```java
+public class Pawn implements Identity.Supplier {
+  private final Location location;
+  private final Identity identity;
+  private final Color    teamColor;
+}
+```
+
+Usually a strategy will specify some desired configuration of agents. This configuration can be
+purely positional, but can also be used for assigning roles to agents (not applicable to chess, 
+since chess roles are not dynamic). For example:
+
+```java
+public class ChessFormation implements Formation<Pawn, Location> {
+  private final Map<Pawn, Location> desiredFormation;
+  
+  @Override
+  public Location getFormationFor(final Pawn pawn) {
+    return this.desiredFormation.get(pawn);
+  }
+}
+```
+
+### Deducers
+
+Incoming data is not always expressed in a compact form such as the snippets described in the system
+overview. When creating game agents using pre-existing protocols, or external systems, the provided
+data representations are often bloated or contain data in undesirable formats. Similarly, data which
+has been received may need to be embellished with information which can be used downstream. 
+This pattern of transforming emitted data is so prevalent that it has been explicitly typed in 
+`zosma` under the name `Deducer`. Note that `Deducer` is simply an interface
+for an [Rx operator](http://reactivex.io/documentation/operators.html):
+
+```java
+public class ChessStrategyDeducer implements Deducer<ChessGame, ChessStrategy> {
+  @Override
+  public Publisher<ChessStrategy> apply(final Publisher<ChessGame> chessGamePublisher) {
+    // compute strategy
+  }    
+}
+```
+
+Which would then be used as follows:
+
+```java
+Flux.from(chessGamePublisher)
+    .transform(new ChessFormationDeducer())
+```
+
+#### Parallelizing deducers
+
+At any point during data processing it might be desirable to parallelize certain computations and
+combine their results. `zosma` contains a `ParallelDeducer` which allow the most recent emissions
+of a list of deducers to be combined:
+
+```java
+Flux.from(chessGamePublisher)
+    .transform(
+        new ParallelDeducer<>(
+            new ChessStrategyDeducer(),
+            new OtherChessStrategyDeducer(),
+            (chessStrategy, otherChessStrategy) -> {
+              // combine, pick, or create a new strategy and return it
+            }))
+```
+
+As can be seen in the example above, `ParallelDeducer` takes deducers as its first *n* arguments 
+(n <= 10) and a combining function as its last argument. The combining function takes *n* arguments, 
+where each argument corresponds to the latest emission by the *n*th provided deducer.
+
+### Putting it together
+
+> graphic of how games are modelled with zosma goes here
+
+`zosma` provides a helper class named `Zosma` for modeling game IO in five parts which represents the
+run configuration of the system:
+
+  1. Publisher of external input
```

Comment:
A list with only ones?

Reply:
Markdown automatically parses this into a numbered list; allowing you to change the order without manually changing the numbers.

Reply:
Ok. So restarting a list is impossible? On atom/vscode this is done by starting with `1.` and following with asterisks. I noticed this doesn't work on github, so I figured it was impossible.